libs/libcommon/src/libcommon/utils.py

```diff
@@ -144,3 +147,32 @@ def is_image_url(text: str) -> bool:
     is_url = text.startswith("https://") or text.startswith("http://")
     (mime_type, _) = mimetypes.guess_type(text.split("/")[-1].split("?")[0])
     return is_url and mime_type is not None and mime_type.startswith("image/")
+
+
+def raise_if_blocked(
+    dataset: str,
+    blocked_datasets: list[str],
+) -> None:
+    """
+    Raise an error if the dataset is in the list of blocked datasets
+
+    Args:
+        dataset (`str`):
+            A namespace (user or an organization) and a repo name separated
+            by a `/`.
+        blocked_datasets (`list[str]`):
+            The list of blocked datasets. If empty, no dataset is blocked.
+            Unix shell-style wildcards are supported, e.g. "open-llm-leaderboard/*"
+
+    Returns:
+        `None`
+    Raises the following errors:
+        - [`libcommon.exceptions.DatasetInBlockListError`]
+          If the dataset is in the list of blocked datasets.
+    """
+    for blocked_dataset in blocked_datasets:
```

Comment:
Like in other PRs, what if by mistake we add "*" to the blocket_datasets? Maybe add a validation here?

Reply:
ahah, good point.

Reply:
https://github.com/huggingface/datasets-server/pull/1899/commits/60645e6800405b712af53f3a55d0a0a62cbaefa8